geoalchemy2/admin/dialects/sqlite.py

```diff
@@ -17,6 +18,21 @@
 from geoalchemy2.types import _DummyGeometry
 
 
+def is_gpkg(bind):
```

Comment:
There are two ways of "identifying" a geopackage now:

- the one over here (by calling the HasGeopackage() and CheckGeoPackageMetaData())
- the one in the `load_spatialite`

For me it has lead to an issue when creating a new geopackage: I make the connection, load the spatialite library myself, then call `gpkgCreateBaseTables` myself, and continue with creating tables. The cached `bind["_is_gpkg"]` is still False in that case, so I run into errors because geoalchemy sees it as a spatialite `RecoverGeometryColumn`.

Why did you opt for not making geopackage a specific dialect? I think it is perfectly fine to expect from users to initiate a dbconn with something like `gpkg:///path/tofile`. Implementation-wise it would clean up a lot also, I expect.

Reply:
I created the new dialect. I had to rework many things but it made the code much better and robust, thanks for the suggestion!

Reply:
Yes it certainly looks a lot better now! Great!